File: go/domain/issue.go
package domain

import (
	"github.com/google/uuid"
)

type Issue struct {
	id          uuid.UUID
	name        string
	description string
}

func (i Issue) GetId() uuid.UUID {
	return i.id
}

func (i Issue) GetName() string {
	return i.name
}

func (i Issue) GetDescription() string {
	return i.description
}

func (i Issue) Update(newName string, newDescription string) Issue {
	if newName != "" {
		i.name = newName
	}
	if newDescription != "" {
		i.name = newDescription
	}
	return i
}

func NewIssue(id uuid.UUID, name string, description string) Issue {
	return Issue{id, name, description}
}

File: node-ts/src/domain/issue.ts
export class Issue {
    constructor(private readonly id: string, 
        private readonly name : string, 
        private readonly description: string){

    }

    getId(): string {
        return this.id
    }

    getName(): string {
        return this.name
    }

    getDescription(): string {
        return this.description
    }

    update(name: string, description: string) : Issue {
        return new Issue(
            this.id,
            Issue.isEmpty(name) ? this.name : name,
            Issue.isEmpty(description) ? this.description : description);
    }
    
    private static isEmpty(str: string): boolean {
        return !str || str.length === 0;
    }
}